doc/man3/OSSL_CMP_CTX_new.pod

```diff
@@ -0,0 +1,654 @@
+stack of untrusted certificates, NULL on error.
+
+OSSL_CMP_CTX_get0_transactionID() returns the transactionID from the context,
+NULL on error.
+
+OSSL_CMP_CTX_get0_newPkey() returns the newPkey from the context,
+or NULL on error.
+
+OSSL_CMP_CTX_get_certConf_cb_arg() returns the certConf callback argument
+set previously, NULL if not set or on function parameter error.
+
+OSSL_CMP_CTX_get_status() returns the PKIStatus from the last CertRepMessage or
+Revocation Response, -1 on error.
+
+OSSL_CMP_CTX_get0_statusString() returns the statusString from the last
+CertRepMessage or Revocation Response, NULL on error.
+
+OSSL_CMP_CTX_get_failInfoCode() returns the failinfo error code bits in context
+as returns bitstring encoded as integer on success, -1 on error.
+
+OSSL_CMP_CTX_get1_extraCertsIn() returns a pointer to a duplicate of the stack
+of X.509 certificates received in the extraCerts field of last received
+certificate response message IP/CP/KUP which had extraCerts set. Returns NULL
+if there were no caPubs as well as on error.
+
+OSSL_CMP_CTX_get1_caPubs() returns a pointer to a duplicate of the stack of
+X.509 certificates received in the caPubs field of last received certificate
+response message IP/CP/KUP which had caPubs set.
+Returns an empty list if there were no caPubs and NULL on error.
+
+All other functions return 1 on success, 0 on error.
+
+=head1 EXAMPLE
+
+The following code does an Initialization Request:
+
+        cmp_ctx = OSSL_CMP_CTX_new();
+        OSSL_CMP_CTX_set1_serverName(cmp_ctx, opt_serverName);
+        OSSL_CMP_CTX_set1_referenceValue(cmp_ctx, ref, ref_len);
+        OSSL_CMP_CTX_set1_secretValue(cmp_ctx, sec, sec_len);
+        OSSL_CMP_CTX_set0_newPkey(cmp_ctx, new_pkey);
+        OSSL_CMP_CTX_set1_caCert(cmp_ctx, ca_cert);
+
+        initialClCert = OSSL_CMP_exec_IR_ses(cmp_ctx);
+
+The following code does an Initialization Request using an
+external identity certificate (RFC 4210, Appendix E.7):
+
+        cmp_ctx = OSSL_CMP_CTX_new();
+        OSSL_CMP_CTX_set1_serverName(cmp_ctx, sname);
+        OSSL_CMP_CTX_set1_clCert(cmp_ctx, cl_cert);
+        OSSL_CMP_CTX_set0_pkey(cmp_ctx, pkey);
+        OSSL_CMP_CTX_set0_newPkey(cmp_ctx, new_pkey);
+        OSSL_CMP_CTX_set1_caCert(cmp_ctx, ca_cert);
+
+        initialClCert = OSSL_CMP_exec_IR_ses(cmp_ctx);
+
+Here externalCert is an X509 certificate granted to the EE by another CA
+which is trusted by the current CA the code will connect to.
+
+
+The following code does a Key Update Request:
+
+        cmp_ctx = OSSL_CMP_CTX_new();
+        OSSL_CMP_CTX_set1_serverName(cmp_ctx, sname);
+        OSSL_CMP_CTX_set0_pkey(cmp_ctx, pkey);
+        OSSL_CMP_CTX_set0_newPkey(cmp_ctx, new_pkey);
+        OSSL_CMP_CTX_set1_clCert(cmp_ctx, cl_cert);
+        OSSL_CMP_CTX_set1_caCert(cmp_ctx, ca_cert);
+
+        updatedClCert = OSSL_CMP_exec_KUR_ses(cmp_ctx);
+
+The following code (which omits error handling) sends a genm including, as an
```

Comment:
genm ?

Reply:
Replaced genm by General Message and genp  by  General Response